pdns/mastercommunicator.cc

```diff
@@ -93,7 +95,7 @@ void CommunicatorClass::queueNotifyDomain(const DNSName &domain, UeberBackend *B
     }
   }
 
-  if (!hasQueuedItem)
+  if (!hasQueuedItem && d_onlyNotify.size())
```

Comment:
OK, NOTIFYs to the NS hosts may be a common use case for Master zones, but not for slave zones. Thus, what about using the zone kind as decision if the warning should be logged? See last commit.